File: app/src/main/java/it/gruppoinfor/home2work/adapters/ShareGuestsAdapter.java
package it.gruppoinfor.home2work.adapters;

import android.app.Activity;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
import com.bumptech.glide.request.RequestOptions;

import java.math.RoundingMode;
import java.text.DecimalFormat;
import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import it.gruppoinfor.home2work.R;
import it.gruppoinfor.home2workapi.model.ShareGuest;

public class ShareGuestsAdapter extends RecyclerView.Adapter<ShareGuestsAdapter.ViewHolder> {


    private Activity activity;
    private ArrayList<ShareGuest> shareGuests;
    private ItemClickCallbacks itemClickCallbacks;

    private DecimalFormat mDf;

    public ShareGuestsAdapter(Activity activity, List<ShareGuest> values) {
        this.activity = activity;
        this.shareGuests = new ArrayList<>(values);
        mDf = new DecimalFormat("#.##");
        mDf.setRoundingMode(RoundingMode.CEILING);
    }

    public void setItemClickCallbacks(ItemClickCallbacks itemClickCallbacks) {
        this.itemClickCallbacks = itemClickCallbacks;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_share_guest, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, final int position) {
        final ShareGuest shareGuest = shareGuests.get(position);

        RequestOptions requestOptions = new RequestOptions()
                .diskCacheStrategy(DiskCacheStrategy.NONE)
                .circleCrop()
                .placeholder(R.drawable.ic_avatar_placeholder);

        Glide.with(activity)
                .load(shareGuest.getGuest().getAvatarURL())
                .transition(DrawableTransitionOptions.withCrossFade())
                .apply(requestOptions)
                .into(holder.userAvatar);


        holder.nameView.setText(shareGuest.getGuest().toString());
        holder.jobView.setText(shareGuest.getGuest().getCompany().toString());

        holder.container.setOnClickListener((v) -> itemClickCallbacks.onItemClick(v, position));
        holder.container.setOnLongClickListener((v) -> itemClickCallbacks.onLongItemClick(v, position));

        if (shareGuest.getStatus() == ShareGuest.Status.COMPLETED) {
            holder.viewStatusCompleted.setVisibility(View.VISIBLE);
        } else if (shareGuest.getStatus() == ShareGuest.Status.CANCELED) {
            holder.viewStatusLeaved.setVisibility(View.VISIBLE);
            holder.itemView.setAlpha(0.7f);
        }

    }

    @Override
    public int getItemCount() {
        return shareGuests.size();
    }


    static class ViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.user_avatar)
        ImageView userAvatar;
        @BindView(R.id.name_view)
        TextView nameView;
        @BindView(R.id.job_view)
        TextView jobView;
        @BindView(R.id.container)
        View container;
        @BindView(R.id.guest_status_completed)
        View viewStatusCompleted;
        @BindView(R.id.guest_status_leaved)
        View viewStatusLeaved;


        ViewHolder(View view) {
            super(view);
            ButterKnife.bind(this, view);
        }
    }
}

File: App/src/main/java/it/gruppoinfor/home2work/adapters/ShareGuestsAdapter.kt
package it.gruppoinfor.home2work.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.bumptech.glide.request.RequestOptions
import it.gruppoinfor.home2work.R
import it.gruppoinfor.home2work.interfaces.ItemClickCallbacks
import it.gruppoinfor.home2workapi.model.Guest
import kotlinx.android.synthetic.main.item_match.view.*
import kotlinx.android.synthetic.main.item_share_guest.view.*
import java.util.*

class ShareGuestsAdapter(private val mContext: Context, values: List<Guest>) : RecyclerView.Adapter<ShareGuestsAdapter.ViewHolder>() {
    private val mGuests: ArrayList<Guest>
    private var mItemClickCallbacks: ItemClickCallbacks? = null

    init {
        mGuests = ArrayList(values)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_share_guest, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val guest = mGuests[position]

        val requestOptions = RequestOptions()
                .diskCacheStrategy(DiskCacheStrategy.NONE)
                .circleCrop()
                .placeholder(R.drawable.ic_avatar_placeholder)

        Glide.with(mContext)
                .load(guest.user.avatarURL)
                .transition(DrawableTransitionOptions.withCrossFade())
                .apply(requestOptions)
                .into(holder.userAvatar)


        holder.nameView.text = guest.user.toString()
        holder.jobView.text = guest.user.company.toString()

        holder.container.setOnClickListener { v -> mItemClickCallbacks!!.onItemClick(v, position) }
        holder.container.setOnLongClickListener { v -> mItemClickCallbacks!!.onLongItemClick(v, position) }

        if (guest.status == Guest.Status.COMPLETED) {
            holder.viewStatusCompleted.visibility = View.VISIBLE
        } else if (guest.status == Guest.Status.CANCELED) {
            holder.viewStatusLeaved.visibility = View.VISIBLE
            holder.itemView.alpha = 0.7f
        }

    }

    override fun getItemCount(): Int {
        return mGuests.size
    }

    fun setItemClickCallbacks(itemClickCallbacks: ItemClickCallbacks) {
        mItemClickCallbacks = itemClickCallbacks
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val nameView: TextView = itemView.share_guest_name_view
        val userAvatar: ImageView = itemView.share_guest_user_avatar
        val jobView: TextView = itemView.share_guest_job_view
        val viewStatusCompleted: View = itemView.guest_status_completed
        val viewStatusLeaved: View = itemView.guest_status_leaved
        val container: View = itemView.share_guest_container
    }
}
